src/components/App.js

```diff
@@ -1,19 +1,306 @@
 import React from 'react';
+import Timer from 'timer-machine';
+import Start from './Start';
+import Card from './Card';
+import Bid from './Bid';
+import RoundResult from './RoundResult';
+import EndScreen from './EndScreen';
 
 import '../styles/components/app.scss';
+import Scoreboard from './Scoreboard';
 
 class App extends React.Component {
-
-  constructor(){
+  constructor() {
     super();
+    this.state = {
+      url: 'https://deckofcardsapi.com/api/deck/new/draw/?count=52',
+      allCards: [],
+      numCards: 3,
+      playerCards: [],
+      computerCards: [],
+      inGame: false,
+      round: 1,
+      roundResult: '',
+      score: {
+        player: 0,
+        computer: 0
+      },
+      winner: '',
+      hiScore: 0,
+      leaderboard: []
+    };
+    this.handleBidClick = this.handleBidClick.bind(this);
+    this.roundAdvance = this.roundAdvance.bind(this);
+    this.startGame = this.startGame.bind(this);
+  }
+
+  // fire the API call on page load
+  componentDidMount() {
+    this.getData(this.state.url);
+  }
+
+  // retrieve a shuffled deck of cards from the API and pass to cleanData processing function
+  getData(url) {
+    fetch(url)
+      .then(response => response.json())
+      .then(data => {
+        this.setState(
```

Comment:
There is too much going on here. Especially with mulitple levels of callbacks. This could be done using a single setState call.
```
const cards = this.state.allCards.concat(data.cards)
const cleanCards = this.cleanData(cards)
this.setState({
  allCards: cleanCards,
  cardsReady: true
});
```